seeds/scripts/test/match-type.test.ts

```diff
@@ -61,6 +61,34 @@ const MATCH_TYPE_CHECKS: Record<
 		},
 	},
 	uscChartHash: { type: "CHARTS", fn: (c) => c.data.hashSHA1 },
+	ddrSongHash: {
+		type: "CHARTS",
+		fn: (c) => {
+			// if there's no ddrSongHash then it must be a konaste song
+			// so just use the inGameID
+			if (c.data.ddrSongHash === undefined) {
+				return `${c.data.inGameID}-${c.difficulty}`;
+			}
+
+			// edge case for the song "B4U (The Acolyte mix)"
+			//
+			// for some reason konmai decided to change its inGameID in DDR A3
+			// so now there's an a20plus version and an a3 version
+			// (see songIDs 37225 and 37310)
+			//
+			// unfortunately, they decided to keep the same ddrSongHash across both!
+			// so we manually distinguish it here.
+			const B4UAcolyteSongHash = "01lbO69qQiP691ll6DIiqPbIdd9O806o";
+			if (c.data.ddrSongHash === B4UAcolyteSongHash) {
+				if (c.versions[0] === "a20plus") {
+					return `${c.data.ddrSongHash}-${c.difficulty}-a20plus`;
+				}
+			}
+			// perhaps we could just merge the two "songs" together in the seeds to avoid all this?
```

Comment:
This has to be handled specially somehow. If we put this in the seeds like this it's a massive landmine for people to step on!

For now, make both of these charts songHash undefined. We'll come back around to fixing it. Not ideal, but not terrible either.